### PR TITLE
dump : fix dump offload related issues to OS

### PR DESCRIPTION
**Pull request by devenrao**

### CODE REVIEW

File: dump/dbus_util.cpp [hunk 1]
```diff
@@ -138,7 +138,7 @@ bool isSystemHMCManaged(sdbusplus::bus::bus& bus)
         return false;
     }
 
-    log<level::INFO>("Util system is not HMC managed");
```
Comment (dhruvibm) [inline dump/dbus_util.cpp:141]: Can you please do just the fix with this commit and other updates please do with separate commit

**Review by dhruvibm — CHANGES_REQUESTED** [review]

File: dump/dbus_util.cpp [hunk 2]
```diff
@@ -160,13 +160,11 @@ bool isHostRunning(sdbusplus::bus::bus& bus)
             return false;
         }
 
-        ProgressStages bootProgess = sdbusplus::xyz::openbmc_project::State::
+        ProgressStages bootProgress = sdbusplus::xyz::openbmc_project::State::
             Boot::server::Progress::convertProgressStagesFromString(*progPtr);
-        if ((bootProgess == ProgressStages::SystemInitComplete) ||
-            (bootProgess == ProgressStages::OSStart) ||
-            (bootProgess == ProgressStages::OSRunning))
+        if (bootProgress == ProgressStages::OSRunning)
```
Comment (deepakala-k) [inline dump/dbus_util.cpp]: I was wondering that in the meanwhile, is someone also working on returning the error from PHYP in case of failure. We might have addressed this particular scenario, but it might be better to address the root cause

File: dump/dbus_util.cpp [hunk 3]
```diff
@@ -138,7 +138,7 @@ bool isSystemHMCManaged(sdbusplus::bus::bus& bus)
         return false;
     }
 
-    log<level::INFO>("Util system is not HMC managed");
+    log<level::INFO>("DumpOffload system is not HMC managed");
```
Comment (Swarnendu-R-C) [inline dump/dbus_util.cpp:144]: At LN128 and LN141 the log messages are only getting changed. Not any functionality. I think this can be made in a separate commit. Just a suggestion

File: dump/dbus_util.cpp [hunk 4]
```diff
@@ -199,7 +198,7 @@ const std::vector<std::string>
         auto response = bus.call(mapperCall);
         response.read(liObjectPaths);
         log<level::INFO>(
-            fmt::format("Util dumps size received is ({}) entry({})",
+            fmt::format("DumpOffload dumps size received is ({}) entry({})",
```
Comment (Swarnendu-R-C) [inline dump/dbus_util.cpp:200]: Same as the comment on LN141

File: dump/host_state_watch.cpp [hunk 5]
```diff
@@ -43,22 +40,18 @@ void HostStateWatch::propertyChanged(sdbusplus::message::message& msg)
                 ProgressStages bootProgress =
                     sdbusplus::xyz::openbmc_project::State::Boot::server::
                         Progress::convertProgressStagesFromString(*progress);
-                if ((bootProgress == ProgressStages::SystemInitComplete) ||
-                    (bootProgress == ProgressStages::OSStart) ||
-                    (bootProgress == ProgressStages::OSRunning))
+                if (bootProgress == ProgressStages::OSRunning)
```
Comment (Swarnendu-R-C) [inline dump/host_state_watch.cpp]: One question, if the BMC Standby is enabled then the OSRunning stage won't be reached. In that case what would be the result? Are we still offloading the file? But by the looking at the code we are not apparently it seems.

**Review by Swarnendu-R-C — COMMENTED** [review]
A few queries and suggestions.

**Review by Swarnendu-R-C — APPROVED** [review]
Looks good to me

File: dump/dbus_util.cpp [hunk 6]
```diff
@@ -146,23 +149,19 @@ bool isHostRunning(sdbusplus::bus::bus& bus)
...
+            sdbusplus::utility::dedup_variant_t<ProgressStages>;
+
+        auto retVal = readDBusProperty<PropertiesVariant>(
+            bus, "xyz.openbmc_project.State.Host",
+            "/xyz/openbmc_project/state/host0",
             "xyz.openbmc_project.State.Boot.Progress", "BootProgress");
-        const std::string* progPtr = std::get_if<std::string>(&retVal);
-        if (progPtr == nullptr)
+        const ProgressStages* progPtr = std::get_if<ProgressStages>(&retVal);
+        if (progPtr != nullptr)
```
Comment (deepakala-k) [inline dump/dbus_util.cpp:160]: Do you want to return if it is not a nullptr?

File: dump/dbus_util.cpp (hunk 2 above)
Comment (devenrao) [reply dump/dbus_util.cpp]: Yeah I tried that at present pldm_send used in this repository is an async call so changed to pldm_send_recv to see if we get any error, if so thought of exploring coroutines, but it was not returning any error. It is simply ignoring the command.

File: dump/host_state_watch.cpp (hunk 5 above)
Comment (devenrao) [reply dump/host_state_watch.cpp]: At BMC standby we do not offload the file, For NON HMC managed systems dumps are offloaded to Operating system. For operating system to start, it need to complete sbe, hostboot ipl and phyp should be loaded and then OS will be loaded. We use the partition in OS to store these dumps. This is the reason we check for OSRunning state.

File: dump/dbus_util.cpp (hunk 6 above)
Comment (devenrao) [reply dump/dbus_util.cpp:157]: this method is called only during startup after that it will depend on property change request that comes to host_state_watch probably the reason this was not breaking but good we fixed

**Review by deepakala-k — APPROVED** [review]

**Review by aravynd — APPROVED** [review]
